### PR TITLE
Added toString implementations for GlobalMemory,VirtulMemory and Firmware

### LINKED ISSUE #945: https://github.com/oshi/oshi/pull/944#issue-306521475
I will work on FirmWare,GlobalMemory and VirtualMemory implementations

_Originally posted by @rohitkukreja1508 in https://github.com/oshi/oshi/issues/942#issuecomment-520208340_

### PR DESCRIPTION
**Pull request by rohitkukreja1508**
Added toString implementations for GlobalMemory,VirtualMemory and Firmware
Fixes #945

### CODE REVIEW

File: oshi-core/src/main/java/oshi/hardware/common/AbstractGlobalMemory.java [hunk 1]
```diff
@@ -37,4 +38,15 @@
     protected long memAvailable = -1L;
     protected long pageSize = -1L;
     protected VirtualMemory virtualMemory;
+    
+    @Override
+    public String toString() {
+    	StringBuilder sb = new StringBuilder();
+    	sb.append("Memory: ");
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/common/AbstractGlobalMemory.java]: Let's leave this "Memory" header off the toString and keep the text in the SystemInfoTest.  And we might want to add the word "Available:" here instead.

File: oshi-core/src/main/java/oshi/hardware/common/AbstractVirtualMemory.java [hunk 2]
```diff
@@ -39,4 +40,15 @@
     protected long swapUsed = -1L;
     protected long swapPagesIn = -1L;
     protected long swapPagesOut = -1L;
+    
+    @Override
+    public String toString() {
+    	StringBuilder sb = new StringBuilder();
+    	sb.append("Swaps used: ");
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/common/AbstractVirtualMemory.java]: Similar to the memory header, let's put "Swap: " in the SystemInfo test and just put "Used:" here instead.

File: oshi-core/src/main/java/oshi/hardware/common/AbstractGlobalMemory.java (hunk 1 above)
Comment (rohitkukreja1508) [reply oshi-core/src/main/java/oshi/hardware/common/AbstractGlobalMemory.java]: Added change for this with "Available" text indented and did the same for "Swap"

**Comment by dbwiddis** [comment]
Looks great!  I'll make some final tweaks and merge.

**Comment by dbwiddis** [comment]
Merging.  Thanks for contributing to open source!

**Pull request merged**